src/services/user/services/AdminUsers.js

```diff
@@ -138,34 +138,131 @@ class AdminUsers {
 		}
 	}
 
-	async create(data, params) {
-		const { account } = params;
-		const currentUserId = account.userId.toString();
-
-		// checks if school isExternal, throws forbidden if it is
+	async create(_data, _params) {
+		const currentUserId = _params.account.userId.toString();
 		const { schoolId } = await getCurrentUserInfo(currentUserId);
+		await this.checkExternal(schoolId);
+		const { email } = _data;
+		await this.checkMail(email);
+		return this.app.service('usersModel').create({
+			..._data,
+			schoolId,
+			roles: [this.role._id],
+		});
+	}
+
+	async update(_id, _data, _params) {
+		if (!_id) throw new BadRequest('id is required');
+		const params = await this.prepareParams(_id, _params);
+		await this.checkExternal(params.query.schoolId);
+		const { email } = _data;
+		await this.checkMail(email, _id);
+		await this.updateAccount(email, _id);
+		// _id is part of params and will be combined with the schoolId
+		const user = await this.prepareRoleback(
+			email, _id, () => this.app.service('usersModel').update(null, _data, params),
+		);
+
+		if (user.length === 0) throw BadRequest('User could not be edit');
+
+		return user;
+	}
+
+	async patch(_id, _data, _params) {
+		if (!_id) throw new BadRequest('id is required');
+		const params = await this.prepareParams(_id, _params);
+		await this.checkExternal(params.query.schoolId);
+		const { email } = _data;
+		await this.checkMail(email, _id);
+		await this.updateAccount(email, _id);
+		// _id is part of params and will be combined with the schoolId
+		const user = await this.prepareRoleback(
+			email, _id, () => this.app.service('usersModel').patch(null, _data, params),
+		);
+
+		if (user.length === 0) throw new BadRequest('user could not be edit');
```

Comment:
```suggestion
		if (user.length === 0) throw new BadRequest('user could not be edited');
```

I think it's weird that `user` is an array here